### PR TITLE
AIML-1395 Move FilterHelper.formatTimestamp to shared util package

### PR DESCRIPTION
**Pull request by ChrisEdwards**
**Jira:** [AIML-1395](https://contrast.atlassian.net/browse/AIML-1395)

## Summary
Move timestamp formatting out of the tool layer into a shared `util` package. 

Also, another PR merged before the archunit tests, so a new violation was added, so this fixed that as well: stop `RecommendationMarkdownRenderer` from importing the SDK `Recommendation` type directly. Both fixes shrink the ArchUnit freeze store by 77 lines with zero added.

- `AttackSummary` and `ServerSummary` no longer depend on `tool.base`
- `RecommendationMarkdownRenderer` takes our own `RecommendationData` type instead of the SDK type
- The result-to-tool dependency cycle is gone

## Why This Change Exists
The ArchUnit tests from AIML-1379 found two problems. `AttackSummary` and `ServerSummary` called `FilterHelper.formatTimestamp` from `tool.base`, so result models depended on tool code. And `RecommendationMarkdownRenderer` accepted `com.contrastsecurity.models.Recommendation` directly, breaking the rule that only `client` and `sdkextension` touch the SDK.

## Approach
**Timestamp formatting.** New `TimestampFormatter` class in a `util` package holds the two formatting methods. Result models import it directly. `FilterHelper` delegates to it so tool-layer callers still work.

**Recommendation type.** New `RecommendationData` record in `sdkextension.data` wraps the three fields the renderer needs, with a `from(Recommendation)` factory. The renderer accepts `RecommendationData` instead of the SDK type. `GetVulnerabilityTool` maps at the call site. This moves the SDK boundary without losing the typed contract.

### New files
- `util/TimestampFormatter.java` — formatting methods extracted from `FilterHelper`
- `sdkextension/data/RecommendationData.java` — record wrapping SDK `Recommendation` fields

### Changed files
- `AttackSummary.java`, `ServerSummary.java` — import `TimestampFormatter` instead of `FilterHelper`
- `FilterHelper.java` — delegates timestamp methods, removed unused constants
- `RecommendationMarkdownRenderer.java` — accepts `RecommendationData`
- `GetVulnerabilityTool.java` — maps through `RecommendationData.from()`

### ArchUnit freeze store
- Cycle-free rule store — emptied (all cycle violations fixed)
- Result-depends-on-tools store — emptied (all violations fixed)
- SDK containment store — 1 stale entry removed

### Harness
- New `archunit` skill with the fix-then-shrink workflow and anti-patterns
- CLAUDE.md trimmed, detail moved to the skill
- Added "Uncommitted Changes" and "Branching and PR Requirements" rules to CLAUDE.md

## Testing
- `TimestampFormatterTest` — 6 tests for both formatters, null handling, edge cases
- `RecommendationDataTest` — 3 tests for field mapping, null input, null fields
- `AttackSummaryTest`, `RecommendationMarkdownRendererTest` — updated to use new types, same behavioral assertions
- 1196 unit tests pass, coverage floors met, PIT mutation testing passes

[AIML-1395]: https://contrast.atlassian.net/browse/AIML-1395?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: contrast-mcp-core/src/test/java/com/contrast/labs/ai/mcp/contrast/sdkextension/data/RecommendationDataTest.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+import com.contrastsecurity.models.Recommendation;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+class RecommendationDataTest {
+
+  @Test
+  void from_should_map_all_fields_from_sdk_recommendation() {
```
Comment (ChrisEdwards) [inline contrast-mcp-core/src/test/java/com/contrast/labs/ai/mcp/contrast/sdkextension/data/RecommendationDataTest.java:27]: I plan to add crap4j to this repo soon, which will allow me to relax test coverage on simple classes like this. It scales the test coverage requirements based on code complexity. Its coming soon.

**Review by Alex-Contrast — APPROVED** [review]
